### PR TITLE
Fullstack/alumni registration and login

### PR DESCRIPTION
**Pull request by reshadbinharun**
Screen Shot 2020-05-18 at 10 07 04 PM

Screen Shot 2020-05-18 at 10 07 32 PM

Screen Shot 2020-05-18 at 10 08 07 PM

** Alumni Registration
*** email validation
*** password matching checking done real-time
*** form validation for required fields

** Login
*** role and userDetails sent from login response are added to store to be accessible from any component in the future!

### CODE REVIEW

File: server/models/alumniSchema.js [hunk 1]
```diff
@@ -5,16 +5,16 @@ const Schema = mongoose.Schema;
...
   {
     name: {type: String, required: true},
-    email: {type: String, reguired: true},
+    email: {type: String, reguired: true, unique: true},
     gradYear: {type: Number, required: true},
-    location: {type: String, required: true},
-    profession: {type: String, required: true},
-    company: {type: String, required: true},
-    college: {type: String, required: true},
+    location: {type: String, required: false},
```
Comment (reshadbinharun) [inline server/models/alumniSchema.js:10]: Made these fields optional!

File: server/routes/alumni.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+              email: email,
+              passwordHash: passwordHash,
+              verificationToken: verificationToken,
+              role: role,
+              emailVerified: emailVerified,
+              approved: approved
+            }
+        );
+        
+        await alumni_instance.save();
```
Comment (reshadbinharun) [inline server/routes/alumni.js:58]: Added awaits on these saves since these are asynchronous operations!

File: client/src/App.js [hunk 3]
```diff
@@ -191,7 +193,7 @@ class App extends Component {
                           navItems={alumniNavBarItems()}
                           activeItem={'home'}
                       />
-                      <div> Home! </div>
+                      <div> Home! Welcome {this.props.details && this.props.details.name} ({this.props.role})</div>
```
Comment (reshadbinharun) [inline client/src/App.js:196]: We are able to use `details` and `role` that was added to `store` in `LoginForm` in this manner!

File: client/src/components/Signup.js [hunk 4]
```diff
@@ -142,50 +240,46 @@ export default class Signup extends React.Component {
...
                         <Form.Field
-                        type="email"
-                        required="true"
-                        style={fieldStyle}
+                            type="email"
+                            required="true"
+                            style={fieldStyle}
+                            error={!this.state.emailValid}
                         >
+                            {!this.state.emailValid ? getErrorLabel('Please enter an invalid email!') : null}
```
Comment (awilli70) [inline client/src/components/Signup.js:248]: nit - should be valid

**Pull request merged**